### DISCUSSION TITLE
Can `output` handle shell functionality like operators or redirects?

### DISCUSSION THREAD

**Discussion by sadamczyk** [twpayne/chezmoi#3085] [General]
I'm trying to handle changes between light/dark mode smoothly, and therefore need to detect which "mode" the device is currently in. I've already got it working on GNOME, but the MacOS command returns an exit code of 1 and dumps output onto stderr as well, if light mode is currently active, and returns a string for dark mode only.

That's why I'm trying to do something like this to avoid the error:
```go
{{- if eq "Dark" (output "defaults" "read" "-g" "AppleInterfaceStyle" "||" "true" | trim) -}}
# dark mode stuff
{{- else -}}
# light mode stuff
{{- end -}}
```

But this doesn't work because the `||` are probably just interpreted as additional arguments passed to `defaults`, instead of the shell operator.
Is there any way to do this, or am I forced to make a shell script for this? I'd like to avoid creating an additional shell script for something so small, and also it means I need to ensure that the shell script exists on the machine _before_ the templates are processed, which also always feels like a bit of a hassle.

**Comment by halostatue** [comment]
You would need to do something like `(output "bash" "-c" "'defaults read -g AppleInterfaceStyle || true'" | trim)`.

**Reply by sadamczyk** [reply]
Ah, of course! I didn't think of that for some reason, thanks a lot!
The single quotes aren't needed btw, they cause an error: `chezmoi: unknown shorthand flag: 'g' in -g`
But just the double quotes work perfectly, as the whole thing gets passed as a string like on the command line.

My final if statement looks like this now:
```go
{{- if eq "Dark" (output "sh" "-c" "defaults read -g AppleInterfaceStyle 2>/dev/null || true" | trim) -}}
```

**Reply by sadamczyk** [reply]
This solution of obtaining dark mode apparently stopped working at some point after MacOS 12 Monterey, which I was running at the time of writing this. (based on a DM I got :))
I don't use a Mac anymore, so I have no idea what the new solution for this is.

Just leaving this here, so anyone reading this isn't confused why this isn't working on their current macOS version. :sweat_smile:

**Reply by halostatue** [reply]
FWIW, I just tested it with 15.5 and it _does_ work, but a value is only returned when the system is in dark mode. It may or may not work with 26. It's very [annoying](https://lmika.org/2022/10/04/the-annoying-way.html).
